src/main/java/meditracker/medication/Medication.java

```diff
@@ -114,30 +151,103 @@ public void setExpiryDate(String expiryDate) {
-        this.remarks = remarks;
+    /**
+     * Checks and sets the remarks. Defaults to Nil if remarks is null.
+     *
+     * @param remarks Remarks value to checked and set
+     */
+    protected void setRemarks(String remarks) {
+        this.remarks = Objects.requireNonNullElse(remarks, "Nil");
     }
 
+    // @@author
+
+    // @@author T0nyLin
+
     public int getRepeat() {
         return repeat;
     }
 
-    public void setRepeat(int repeat) {
+    /**
+     * Checks if repeat is within range of 1 to 7
+     *
+     * @param repeat Repeat value to be checked and set
+     * @throws MediTrackerException When the value is not within the specified range
+     */
+    protected void setRepeat(int repeat) throws MediTrackerException {
+        if (repeat < 1 || repeat > 7) {
+            throw new MediTrackerException("Provide a \"-rep\" number from 1 to 7");
+        }
         this.repeat = repeat;
     }
 
     public int getDayAdded() {
         return dayAdded;
     }
 
-    public void setDayAdded(int dayAdded) {
+    protected void setDayAdded(int dayAdded) {
         this.dayAdded = dayAdded;
     }
 
+    // @@author
+
+    // @@author annoy-o-mus
+
+    /**
+     * Calls the setter method with the parsed data based on the argumentName specified
+     *
+     * @param argumentName ArgumentName that identifies what type of data it is
+     * @param argumentValue Value of the data to be (converted and) set
+     * @throws MediTrackerException If it fails the checks and/or parsing
+     */
+    public void setMedicationValue(ArgumentName argumentName, String argumentValue) throws MediTrackerException {
+        switch (argumentName) {
+        case NAME:
+            setName(argumentValue);
+            break;
+        case QUANTITY:
+            double quantity = convertStringToDouble(argumentValue);
+            setQuantity(quantity);
+            break;
+        case DOSAGE_MORNING:
+            double dosageMorning = convertStringToDouble(argumentValue);
+            setDosageMorning(dosageMorning);
+            break;
+        case DOSAGE_AFTERNOON:
+            double dosageAfternoon = convertStringToDouble(argumentValue);
+            setDosageAfternoon(dosageAfternoon);
+            break;
+        case DOSAGE_EVENING:
+            double dosageEvening = convertStringToDouble(argumentValue);
+            setDosageEvening(dosageEvening);
+            break;
+        case EXPIRATION_DATE:
+            setExpiryDate(argumentValue);
+            break;
+        case REMARKS:
+            setRemarks(argumentValue);
+            break;
+        case REPEAT:
+            int repeat = convertStringToInteger(argumentValue);
+            setRepeat(repeat);
+            break;
+        case DAY_ADDED:
+            int dayAdded = convertStringToInteger(argumentValue);
+            setDayAdded(dayAdded);
+            break;
+        default:
+            throw new IllegalStateException("Unexpected value: " + argumentName);
```

Comment:
Will this cause the program to crash if for some reason this gets thrown? It looks like this exception is unhandled. I am thinking of throwing `MediTrackerException` instead.

Reply:
Resolved in 845916f